Userland/Libraries/LibPDF/Value.cpp

```diff
@@ -9,52 +9,31 @@
 
 namespace PDF {
 
-Value& Value::operator=(Value const& other)
-{
-    m_type = other.m_type;
-    switch (m_type) {
-    case Type::Empty:
-    case Type::Null:
-        break;
-    case Type::Bool:
-        m_as_bool = other.m_as_bool;
-        break;
-    case Type::Int:
-        m_as_int = other.m_as_int;
-        break;
-    case Type::Float:
-        m_as_float = other.m_as_float;
-        break;
-    case Type::Ref:
-        m_as_ref = other.m_as_ref;
-        break;
-    case Type::Object:
-        m_as_object = other.m_as_object;
-        break;
-    }
-    return *this;
-}
-
 String Value::to_string(int indent) const
 {
-    switch (m_type) {
-    case Type::Empty:
-        return "<empty>";
-    case Type::Null:
-        return "null";
-    case Type::Bool:
-        return as_bool() ? "true" : "false";
-    case Type::Int:
-        return String::number(as_int());
-    case Type::Float:
-        return String::number(as_float());
-    case Type::Ref:
-        return String::formatted("{} {} R", as_ref_index(), as_ref_generation_index());
-    case Type::Object:
-        return as_object()->to_string(indent);
-    }
-
-    VERIFY_NOT_REACHED();
+    return visit(
+        [&](Empty const&) {
+            // Return type deduction means that we can't use implicit conversions.
```

Comment:
You can also use a trailing return type for the lambda.

Reply:
Oh, nice! C++ sure has some fancy features :D